modules/core/shared/src/main/scala/lucuma/core/math/ProperMotion.scala

```diff
@@ -32,36 +32,40 @@ final case class ProperMotion(
   dec: ProperMotion.AngularVelocityComponent[VelocityAxis.Dec]
 ) {
   // Return the ra/dec components in radians, first converting to degrees/y
-  def toRadians: (Double, Double) =
+  inline def toRadians: (Double, Double) =
     (ra.toRadians, dec.toRadians)
 
 }
 
 object ProperMotion extends ProperMotionOptics {
-  final case class AngularVelocityComponent[A](μasy: Quantity[Long, MicroArcSecondPerYear]) {
+  opaque type AngularVelocityComponent[A] = Quantity[Long, MicroArcSecondPerYear]
+  extension[A](self: AngularVelocityComponent[A])
+    inline def μasy: Quantity[Long, MicroArcSecondPerYear] = self
+
     // Direct conversion via coulomb turns to be too slow
-    def toRadians: Double = (μasy.value.toDouble / (3600 * 1e6)).toRadians
+    inline def toRadians: Double = (μasy.value.toDouble / (3600 * 1e6)).toRadians
 
-    val masy: Quantity[Rational, MilliArcSecondPerYear] = μasy.toValue[Rational].toUnit[MilliArcSecondPerYear]
+    def masy: Quantity[Rational, MilliArcSecondPerYear] = μasy.toValue[Rational].toUnit[MilliArcSecondPerYear]
 
-    override def toString =
+    def toString: String =
       s"AngularVelocityComponent(${masy.show})"
-  }
 
   object AngularVelocityComponent extends AngularVelocityComponentOptics {
+    inline def apply[A](w: AngularVelocityComponent[A]): AngularVelocityComponent[A] = w
```

Comment:
Is this used to build the opaque type? Should the parameter `w` be a `Quantity[Long, MicroArcSecondPerYear]` ?

Reply:
I think you're right. This works here because they are in the same file. I'll change it